### PR TITLE
Add umap-learn package

### PR DESCRIPTION
**Pull request by lmcinnes**
First pass -- ideally this should be relatively straightforward as there are no build requirements.

### CODE REVIEW

File: recipes/umap-learn/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    - python
+    - numpy >=1.13
+    - numba >=0.35
+    - scikit-learn >=0.19
+    - scipy >=0.18
+
+test:
+  imports:
+    - umap
+  requires:
```
Comment (synapticarbors) [inline recipes/umap-learn/meta.yaml]: For a simple import test, you don't need to require `nose` as far as I can tell. Please remove this if that's the case.

File: recipes/umap-learn/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+build:
+  number: 0
+  skip: True  # [win and py27 or win32]
+  script: python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  build:
+    - python
+    - pip
+  run:
```
Comment (synapticarbors) [inline recipes/umap-learn/meta.yaml]: Do we need `setuptools` as a run requirement due to using `pkg_resources`?

https://github.com/lmcinnes/umap/blob/master/umap/__init__.py#L3

File: recipes/umap-learn/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  imports:
+    - umap
+  requires:
+    - nose
+
+about:
+  home: http://github.com/lmcinnes/umap
+  license: BSD-2-Clause
+  license_family: BSD
+  # license_file: LICENSE.txt
```
Comment (synapticarbors) [inline recipes/umap-learn/meta.yaml]: Please manually package the license and consider adding it to the source distribution:

https://conda-forge.org/docs/meta.html#packaging-the-license-manually

File: recipes/umap-learn/meta.yaml (hunk 1 above)
Comment (lmcinnes) [reply recipes/umap-learn/meta.yaml]: How do I run the nose tests themselves? I am a bit of a newbie at the whole conda packaging system, so I'm not entirely sure what the right syntax should be.

File: recipes/umap-learn/meta.yaml (hunk 2 above)
Comment (lmcinnes) [reply recipes/umap-learn/meta.yaml]: Yes, I believe we do (it seemed to work anyway somehow, but explicit is better than implicit).

File: recipes/umap-learn/meta.yaml (hunk 1 above)
Comment (lmcinnes) [reply recipes/umap-learn/meta.yaml]: In fact it doesn't matter; the tests aren't packaged by PyPI (I'll have to fix that in the future) so they aren't available. Removing nose.

Comment (synapticarbors) [reply recipes/umap-learn/meta.yaml]: Typically we don't require you to run the full test suite, but if it doesn't take too long to run and you want to do it, there are a variety of ways. For straight-up unittests, I've done:

```
commands:
    - python -m unittest discover -s tests --verbose
```

You can do something similar with the `nosetests ` command as you do in your own tests. Then you'd want to have `nose` as a `test` requirement. This all assumes you distribute the test modules with the source distribution.

**Pull request merged**